src/peer.rs

```diff
@@ -76,6 +46,15 @@ pub struct Peer {
 }
 
 impl Peer {
+    /// Constructs a [Peer] from the given configuration and handlers, and start the thread for the p2p network.
+    pub async fn start(config: Config, handlers: Vec<Box<dyn Fn(PublicAddress, Message) + Send>>) -> Peer {
```

Comment:
@AlvinHon please advise if we need to handle the error, like return Result<Peer, EngineStartError> or this is just fine

Reply:
From user's perspective, engine is not visible to user, but `Peer::start`. `PeerStartError` could be more appropriate.